packages/core/test/unit/dialects/mssql/connection-manager.test.js

```diff
@@ -43,7 +43,7 @@ if (dialect === 'mssql') {
       this.connectionStub.restore();
     });
 
-    it('connectionManager._connect() does not delete `domain` from config.dialectOptions', async function () {
+    it.skip('connectionManager._connect() does not delete `domain` from config.dialectOptions', async function () {
```

Comment:
The tests have been skipped as the lib is now a private value with cannot be accessed and stubbed.
Are these still required?

Reply:
First test looks like a test that was added due to a bug. Ideally should be kept. The lib could be a TypeScript private property instead of a JS private. We'll also probably re-add support for the `dialectModule` option in a cleaner way, so that option will be usable in the future to provide a stub

Second test is incorrectly written as, if the test does not throw, the test will succeed incorrectly. If it turns out it already never throws, can be removed

Third test doesn't make sense to me: It doesn't matter if a tedious API was called when establishing the connection. All that matters is that it was established. Could be removed

Reply:
I've made the `lib` a private property and converted the test to typescript and re-enabled all the tests.
The last test is ensuring the we call the `connect` method if we can an initialized connection from the new `Connection` class.